modules/collection/desktops/niri.nix

```diff
@@ -0,0 +1,222 @@
+      commands:
+        "spawn-at-startup " + (toNiriSpawn commands)
+    )
+    spawn;
+
+  niriEnvironment = let
+    variables = pipe (config.environment.sessionVariables // cfg.extraVariables) [
+      (mapAttrsToList (n: v: n + " \"${v}\""))
+      (concatStringsSep "\n")
+    ];
+  in ''
+    environment {
+      ${variables}
+    }
+  '';
+
+  bindsModule = submodule {
+    options = {
+      spawn = mkOption {
+        type = str;
+        default = null;
+        example = ["foot" "-e" "fish"];
+        description = ''
+          [niri's wiki]: https://github.com/YaLTeR/niri/wiki/Configuration:-Key-Bindings
+
+          The spawn action to run on button-press. For other actions, please see
+          {option}`binds.<keybind>.actions`. See [niri's wiki] for more information.
+        '';
+      };
+      action = mkOption {
+        type = str;
+        default = null;
+        example = "focus-column-left";
+        description = ''
+          [niri's wiki]: https://github.com/YaLTeR/niri/wiki/Configuration:-Key-Bindings
+
+          The non-spawn action to run on button-press. For spawning processes, please see
+          {option}`binds.<keybind>.spawn`. See [niri's wiki] for a complete list.
+        '';
+      };
+      parameters = mkOption {
+        type = attrsOf anything;
+        default = {};
+        example = {
+          allow-when-locked = true;
+          cooldown-ms = 150;
+        };
+        description = ''
+          [niri's wiki]: https://github.com/YaLTeR/niri/wiki/Configuration:-Key-Bindings
+
+          The paramaters to append to the bind. See [niri's wiki] for a complete list.
+        '';
+      };
+    };
+  };
+
+  cfg = config.rum.desktops.niri;
+in {
+  options.rum.desktops.niri = {
+    enable = mkEnableOption "niri: A scrollable-tiling Wayland compositor";
+    binds = mkOption {
+      type = attrsOf bindsModule;
+      default = {};
+      example = {
+        "Mod+Return" = {
+          spawn = ["foot" "-e" "fish"];
+          parameters = {
+            allow-when-locked = true;
+            cooldown-ms = 150;
+          };
+        };
+        "Mod+D" = {
+          action = "close-window";
+          parameters = {
+            repeat = false;
+          };
+        };
+      };
+    };
+    spawn-at-startup = mkOption {
+      type = listOf (listOf str);
+      default = [];
+      example = literalExpression ''
+        [
+          ["waybar"]
+          ["${pkgs.alacritty}/bin/alacritty" "-e" "fish"]
+        ]
+      '';
+    };
+    extraVariables = mkOption {
+      type = attrsOf str;
+      default = {};
+      example = {
+        DISPLAY = ":0";
+      };
+      description = ''
+        Extra environmental variables to be added to Niri's `enviroment` node.
+        This can be used to override variables set in {option}`enviroment.sessionVariables`.
+        You can therefore set a variable to `null` to force unset it in Niri.
+      '';
+    };
+    configFiles = mkOption {
```

Comment:
Why wouldn't we have `configFile` here? One can use `pkgs.concatText` themselves here, and I think we should rather enforce a single file, as Niri really just reads from `config.kdl` (which would align with [their documentation](https://github.com/YaLTeR/niri/wiki/Configuration:-Introduction#loading) too.